examples/micro-stats.c

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "examples.h"
+
+typedef struct service_state_s
+{
+    // in a real application this should be protected by a mutex. In this
+    // example, the main control flow provides synchronization.
+    int odd_count;
+} service_state_t;
+
+static microError *
+handle_default(microRequest *req)
+{
+    char buf[64];
+    const char *response = "odd";
+    int n;
+    service_state_t *state = microRequest_GetServiceState(req);
+
+    snprintf(buf, sizeof(buf), "%.*s", microRequest_GetDataLength(req), microRequest_GetData(req));
+    n = atoi(buf);
+    if (n % 2 != 0)
+    {
+        // this should be protected by a mutex in a real application.
+        state->odd_count++;
+
+        response = "even";
+    }
+    return microRequest_Respond(req, response, strlen(response));
+}
+
+static microError *
+handle_stats(microRequest *req)
+{
+    microError *err = NULL;
+    microServiceStats *stats = NULL;
+    char buf[2048];
+    service_state_t *service_state = microRequest_GetServiceState(req);
+    int total, custom, len;
+
+    err = microService_GetStats(&stats, microRequest_GetService(req));
+    if (err != NULL)
+        return err;
+
+    total = stats->Endpoints[0].num_requests;
```

Comment:
👍 What do you use to (cross?-)compile on Windows? I don't have a local setup and the CI doesn't cover it. Or should I get a Windows/Parallels box as my dev environment for that?

Reply:
I have a VirtualBox with Windows and I just use nmake since our CMakeFile takes care of cross compiling. I believe there is an example in the README of NATS C client how to generate the windows CMake env to use NMake.